sql/core/src/main/scala/org/apache/spark/sql/GroupedData.scala

```diff
@@ -39,13 +39,14 @@ class GroupedData protected[sql](df: DataFrameImpl, groupingExprs: Seq[Expressio
       df.sqlContext, Aggregate(groupingExprs, namedGroupingExprs ++ aggExprs, df.logicalPlan))
   }
 
-  private[this] def aggregateNumericColumns(f: Expression => Expression): Seq[NamedExpression] = {
-    df.numericColumns.map { c =>
-      val a = f(c)
-      Alias(a, a.toString)()
-    }
+  private[this] def aggregateNumericColumns(colNames: String*)
```

Comment:
btw i just wrote the above in an editor. haven't tried compiling it it so it probably wouldn't compile.